packages/components/src/public/QueryModel/QueryModel.ts

```diff
@@ -765,6 +766,13 @@ export class QueryModel {
         return this.selections?.size > 0;
     }
 
+    /**
+     * Key to attach to selections, which are specific to a view
+     */
+    get selectionKey(): string {
+        return this.id + (this.viewName ? '/' + encodePart(this.viewName).toLowerCase() : '');
```

Comment:
Great to see that this could be the main/one place to add this to the key and have the selection actions use it.